internal/csi-addons/rbd/replication_test.go

```diff
@@ -438,58 +440,118 @@ func TestCheckRemoteSiteStatus(t *testing.T) {
-			nil,
-			"failed to unmarshal",
+			name: "description without local_snapshot_time",
+			//nolint:lll // sample output cannot be split into multiple lines.
+			description: `replaying, {"bytes_per_second":0.0,"bytes_per_snapshot":81920.0,"last_snapshot_bytes":81920,"last_snapshot_sync_seconds":56743,"remote_snapshot_timestamp":1684675261,"replay_state":"idle"}`,
+			info: localstruct{
+				lastsyncduration: nil,
+				lastsynctime:     nil,
+				lastsyncbytes:    0,
+			},
+			expectedErr: corerbd.ErrLastSyncTimeNotFound.Error(),
 		},
 		{
-			"description with no JSON",
-			`replaying`,
-			nil,
-			corerbd.ErrLastSyncTimeNotFound.Error(),
+			name: "description without last_snapshot_sync_seconds",
+			//nolint:lll // sample output cannot be split into multiple lines.
+			description: `replaying, {"bytes_per_second":0.0,"bytes_per_snapshot":81920.0,"last_snapshot_bytes":81920,"local_snapshot_timestamp":1684675261,"remote_snapshot_timestamp":1684675261,"replay_state":"idle"}`,
+			info: localstruct{
+				lastsyncduration: nil,
+				lastsynctime:     timestamppb.New(time.Unix(1684675261, 0)),
+				lastsyncbytes:    81920,
+			},
+			expectedErr: "",
+		},
+		{
+			name: "description with invalid JSON",
+			//nolint:lll // sample output cannot be split into multiple lines.
+			description: `replaying,{"bytes_per_second":0.0,"last_snapshot_bytes":81920","bytes_per_snapshot":149504.0","remote_snapshot_timestamp":1662655501`,
+			info: localstruct{
+				lastsyncduration: nil,
+				lastsynctime:     nil,
+				lastsyncbytes:    0,
+			},
+			expectedErr: "failed to unmarshal",
+		},
+		{
+			name:        "description with no JSON",
+			description: `replaying`,
+			info: localstruct{
+				lastsyncduration: nil,
+				lastsynctime:     nil,
+				lastsyncbytes:    0,
+			},
+			expectedErr: corerbd.ErrLastSyncTimeNotFound.Error(),
 		},
 	}
 	for _, tt := range tests {
 		tt := tt
 		t.Run(tt.name, func(t *testing.T) {
 			t.Parallel()
-			ts, err := getLastSyncTime(tt.description)
+			teststruct, err := getLastSyncInfo(tt.description)
 			if err != nil && !strings.Contains(err.Error(), tt.expectedErr) {
 				// returned error
-				t.Errorf("getLastSyncTime() returned error, expected: %v, got: %v",
+				t.Errorf("getLastSyncInfo() returned error, expected: %v, got: %v",
 					tt.expectedErr, err)
 			}
-			if !ts.AsTime().Equal(tt.timestamp.AsTime()) {
-				t.Errorf("getLastSyncTime() %v, expected %v", ts, tt.timestamp)
+			if teststruct.lastsynctime.GetSeconds() != tt.info.lastsynctime.GetSeconds() {
+				t.Errorf("name: %v, getLastSyncInfo() %v, expected %v", tt.name, teststruct.lastsynctime, tt.info.lastsynctime)
+			}
+			if teststruct.lastsyncduration.GetSeconds() != tt.info.lastsyncduration.GetSeconds() {
+				t.Errorf("name: %v, getLastSyncInfo() %v, expected %v", tt.name, teststruct.lastsyncduration,
+					tt.info.lastsyncduration)
+			}
```

Comment:
```suggestion
			if teststruct.lastsyncduration == nil &&  tt.info.lastsyncduration !=nil {
				t.Errorf("name: %v, getLastSyncInfo() %v, expected %v", tt.name, teststruct.lastsyncduration,
					tt.info.lastsyncduration)
			}
			if teststruct.lastsyncduration.GetSeconds() != tt.info.lastsyncduration.GetSeconds() {
				t.Errorf("name: %v, getLastSyncInfo() %v, expected %v", tt.name, teststruct.lastsyncduration,
					tt.info.lastsyncduration)
			}
```

Add another check here, 
a nil duration GetSeconds is giving 0 that is why the `missing last_sync_duration ` case is not getting caught here.

Reply:
Couldn't get the purpose of the extra check, can you please elaborate?
If the last_sync_duration is missing, the function returns nil and so does the testcase has and GetSeconds of both will be 0 and if it is not, the error will be caught.

Reply:
Have you checked it with a println function ? 

teststruct.lastsyncduration.GetSeconds() gives a 0 when teststruct.lastsyncduration is nil.

Reply:
yes, I know that, but what is the issue with it?